src/map/besieged_system.cpp

```diff
@@ -21,105 +21,80 @@ along with this program.  If not, see http://www.gnu.org/licenses/
-    {
-        return 0;
-    }
-
-    uint8 GetMamookPrisoners()
-    {
-        return 0;
-    }
-
-    bool GetMamookMirrorDestroyed()
-    {
-        return 0;
-    }
-
-    uint8 GetHalvungOrders()
-    {
-        return ORDERS_DEFEND;
-    }
-
-    uint8 GetHalvungLevel()
-    {
-        return STRONGHOLD_LEVEL::LEVEL1;
-    }
-
-    uint8 GetHalvungForces()
-    {
-        return 95;
-    }
-
-    uint8 GetHalvungMirrors()
-    {
-        return 0;
-    }
-
-    uint8 GetHalvungPrisoners()
-    {
-        return 0;
-    }
-
-    bool GetHalvungMirrorDestroyed()
-    {
-        return 0;
-    }
-
-    uint8 GetArrapagoOrders()
-    {
-        return ORDERS_DEFEND;
-    }
-
-    uint8 GetArrapagoLevel()
-    {
-        return STRONGHOLD_LEVEL::LEVEL1;
-    }
-
-    uint8 GetArrapagoForces()
-    {
-        return 95;
-    }
-
-    uint8 GetArrapagoMirrors()
-    {
-        return 0;
-    }
-
-    uint8 GetArrapagoPrisoners()
-    {
-        return 0;
-    }
-
-    bool GetArrapagoMirrorDestroyed()
-    {
-        return 0;
-    }
-}; // namespace besieged
+    static std::shared_ptr<BesiegedData> besiegedData;
+
+    std::shared_ptr<BesiegedData> GetBesiegedData()
+    {
+        if (besiegedData == nullptr)
+        {
+            besiegedData = std::make_shared<BesiegedData>(sql);
+        }
+
+        return besiegedData;
+    }
+
+    /**
+     * Called when stronghold updates are received from the world server
+     */
+    void HandleStrongholdUpdate(std::vector<stronghold_info_t> const& strongHoldInfos)
+    {
+        TracyZoneScoped;
+
+        if (settings::get<bool>("logging.DEBUG_BESIEGED"))
+        {
+            ShowInfo("Received besieged Stronghold Data:");
+            for (uint8 strongholdId = 0; strongholdId < strongHoldInfos.size(); strongholdId++)
+            {
+                stronghold_info_t strongholdInfo = strongHoldInfos[strongholdId];
+                ShowInfo("Stronghold %d: Orders %d, Level %d, Forces %d, Mirrors %d, Prisoners %d, Mirror Destroyed %d, Owns Astral Candescence %d",
+                         strongholdId, strongholdInfo.orders, strongholdInfo.stronghold_level, strongholdInfo.forces, strongholdInfo.mirrors, strongholdInfo.prisoners, strongholdInfo.mirrorDestroyed, strongholdInfo.ownsAstralCandescence);
+            }
+        }
+
+        GetBesiegedData()->updateStrongholdInfos(strongHoldInfos);
+    }
+
+    /**
+     * HandleZMQMessage is called by message_server when a besieged ZMQ message is receieved
+     */
+    void HandleZMQMessage(uint8* data)
+    {
+        uint8 subtype = ref<uint8>(data, 1);
+        switch (subtype)
+        {
+            case BESIEGED_WORLD2MAP_STRONGHOLD_INFO:
+            {
+                const std::size_t              headerLength    = 2 * sizeof(uint8);
+                std::size_t                    size            = ref<std::size_t>(data, 2);
+                std::vector<stronghold_info_t> strongholdInfos = std::vector<stronghold_info_t>(size);
```

Comment:
For long types like this, use `auto` so it looks sane

Reply:
Done